src/server/http/xsrf.js

```diff
@@ -4,18 +4,49 @@ export default function (kbnServer, server, config) {
   const disabled = config.get('server.xsrf.disableProtection');
   const versionHeader = 'kbn-version';
   const xsrfHeader = 'kbn-xsrf';
+  const contentTypeHeader = 'content-type';
+  const allowedRequestMediaTypes = ['application/json', 'application/x-ndjson'];
 
   server.ext('onPostAuth', function (req, reply) {
     if (disabled) {
       return reply.continue();
     }
 
     const isSafeMethod = req.method === 'get' || req.method === 'head';
-    const hasVersionHeader = versionHeader in req.headers;
-    const hasXsrfHeader = xsrfHeader in req.headers;
+    if (!isSafeMethod) {
```

Comment:
The only reason I didn't want to do that because we'll have 3 exit points instead of 2: two `reply.continue()` (one additional for early return) and one for bad request. Is it ok for you?